### ISSUE TITLE
ClusterMotif execution using test dataset: error related to gimmemotifs run

### ISSUE THREAD

**Issue by deep-buddingcoder** [loosolab/TOBIAS#207]
Hi Mette,

I encountered a problem while trying to test run "ClusterMotifs" using test data set provided at https://github.com/loosolab/TOBIAS/wiki/installation.

Linux session is mentioned below:

$ TOBIAS ClusterMotifs --motifs test_data/motifs.jaspar --threshold 0.4 --type png
# TOBIAS 0.13.3 ClusterMotifs (run started 2023-05-01 10:51:24.516707)
.
.
.
# color:        YlOrRd_r

Traceback (most recent call last):
  File "/home/deep/miniconda3/envs/tobias_DNAFootPrinting/bin/TOBIAS", line 33, in <module>
    sys.exit(load_entry_point('tobias==0.13.3', 'console_scripts', 'TOBIAS')())
  File "/home/deep/miniconda3/envs/tobias_DNAFootPrinting/lib/python3.6/site-packages/tobias/TOBIAS.py", line 154, in main
    args.func(args)
  File "/home/deep/miniconda3/envs/tobias_DNAFootPrinting/lib/python3.6/site-packages/tobias/tools/motif_clust.py", line 305, in run_motifclust
    gimme_version = gimmemotifs.__version__
AttributeError: module 'gimmemotifs' has no attribute '__version__'

$

gimmemotifs was re-installed using conda command but that did not solve the problem.

Question1: Will it be possible to let me know what version of gimmemotifs was used in TOBIAS?

Question2: I could not make proper sense of the error, will it be possible to decipher the error?

Question3: Will it be possible to create "tobias_env.txt" file of "tobias_env" (equivalent to tobias_env.yaml) and share it in github?
hint: conda list --explicit > tobias_env.txt
Reference: https://conda.io/projects/conda/en/latest/user-guide/tasks/manage-environments.html
ENV*.txt file contains greater details compared to ENV*.yaml file.

I will appreciate your insight/advice.
Thanks.

Special note: Everything else ran w/o encountering any error using the test data set.

**Comment by msbentsen** [comment]
Hi,

Can you try to update your tobias installation? For example with:
`pip install tobias --upgrade`

The line giving the error is different in the current version, so the issue might already have been solved. Reference:
https://github.com/loosolab/TOBIAS/blob/3c93332ea33502b8dd7b04dee313ebb0740f2f40/tobias/tools/motif_clust.py#L305

**Comment by deep-buddingcoder** [comment]
Hi Mette,

Thanks for your response and advice. 

I did upgrade tobias but it did not solve the issue related to gimmemotif as the prior installed version was 0.11.
I also installed gimmemotifs=0.17.0 using conda command, but gimme motifs could not be called while running ClusterMotifs.
The last two lines of the error I received was:

`File "/home/deep/miniconda3/envs/tobias_DNAFootPrint/lib/python3.6/site-packages/genomepy/caching.py", line 33, in <module>
    @lock
TypeError: 'UnixFileLock' object is not callable`
`
Nevertheless. I have by-passed the issue by installing TOBIAS_snakemake version. The test run worked for the command option as mentioned below:
$ snakemake --configfile example_config.yaml --use-conda --cores 10 --conda-prefix /tmp --rerun-incomplete

Thanks
Deep
